Sources/SPMBuildCore/BuildSystem/BuildSystem.swift

```diff
@@ -59,11 +60,20 @@ public protocol BuildSystem: Cancellable {
 
 extension BuildSystem {
     /// Builds the default subset: all targets excluding tests.
-    public func build() async throws {
+    @discardableResult
+    public func build() async throws -> BuildResult {
         try await build(subset: .allExcludingTests)
     }
 }
 
+public struct BuildResult {
+    package init(serializedDiagnosticPathsByTargetName: Result<[String: [AbsolutePath]], Error>) {
+        self.serializedDiagnosticPathsByTargetName = serializedDiagnosticPathsByTargetName
+    }
+    
+    public var serializedDiagnosticPathsByTargetName: Result<[String: [AbsolutePath]], Error>
```

Comment:
I think there is a problem with using a "name" here as a string unless we can 100% make sure that it would distinguish between host and target builds of the same module (i.e. in BuildOperation we use `module.name` which won't make a distinction)...

Reply:
I added another test here to ensure only one fix-it is applied even if the sources need to be built twice as a common dependency of host tools and target content. I think using the name here is in line with the fact that the list of targets may be user provided - names uniquely identify targets in the manifest model, just not once they've been lowered to targets building for a specific platform

Reply:
Right now they'd be concatenated instead of one overriding the other, so the entry corresponding to "Foo" includes the diagnostic paths for Foo when built for the host and those when built for the target, and we rely on SwiftFixIt to deduplicate them. I'm not sure we want to drop the host diagnostics entirely because that feels arbitrary imo.

Reply:
Oh I see, yeah, that sounds okay, SwiftFixIt should be able to handle that!